source/credentials_provider_sts_web_identity.c

```diff
@@ -95,12 +106,8 @@ static void s_user_data_destroy(struct sts_web_identity_user_data *user_data) {
         impl->function_table->aws_http_connection_manager_release_connection(
             impl->connection_manager, user_data->connection);
     }
-
+    s_user_data_reset_request_and_response(user_data);
```

Comment:
reset won't clean up buf, just reset len and ptr, so as to avoid buf alloc/dealloc multiple times. Then we need clean up the buf during destroy